app/org/sagebionetworks/bridge/services/SurveyService.java

```diff
@@ -310,7 +310,8 @@ private void checkConstraintsBeforePhysicalDelete(final StudyIdentifier studyId,
             return surveyReference.getGuid().equals(theseKeys.getGuid());
         });
         if (match != null) {
-            long publishedSurveys = getSurveyAllVersions(studyId, keys.getGuid()).stream()
+            // If you delete a version, things that depend on it are eligible for physical deletion.
+            long publishedSurveys = getSurveyAllVersions(studyId, keys.getGuid(), false).stream()
```

Comment:
Should we update SurveyServiceTest.deleteSurveyPermanentlyConstrainedByScheduleWithPublishedSurvey() to make sure that logically deleted versions don't count against this threshold?

Reply:
Added a couple of more tests to verify that a call with one published survey throws, but a survey with older published versions does not. The call doesn't return logically deleted versions but the logic in the service is correct.